### ISSUE TITLE
v12 - process.env variables are not available at forRoot time in nestjs

### ISSUE THREAD

**Issue by yharaskrik** [nrwl/nx#5692]
## Current Behavior

When building and running a Nestjs app with `node main.js` the process.env variables are not available during `AppModule` instantiation time but are during `forRootAsync`. We are using `dotenv-safe` to load our `.env` file and pass a `configuration` function into the `ConfigModule.forRoot`. This was working prior to upgrade to the version of Nx that upgrades angular to v12 (i thnk that would be Nx 12.3.0 but i could be wrong).

Note: The process.env forRoot method does work if just serving with `yarn nx serve app` it is only when building the assets and running the compiled `main.js` file.

I realize this may not actually be an Nx issue itself but I am not positive, this may or may not be in the nrwl builder

## Expected Behavior

Environment files should work the same way as they did before.

## Steps to Reproduce

1. Generate new latest workspace with Nest preset
2. add `dotenv-safe`
3. create new `.env` and `.env.example` and add a test var
4. add `require('dotenv-safe').config({ allowEmptyValues: true });` to main.ts
5. log the `process.env.TEST_VAR` in appmodule
6. Serving will show the value in the console, building and running `node main.js` will log `undefined`

### Environment

```
 NX  Report complete - copy this into the issue template

  Node : 12.20.1
  OS   : darwin x64
  npm  : 6.14.10
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 12.3.3
  @nrwl/cypress : Not Found
  @nrwl/devkit : 12.3.3
  @nrwl/eslint-plugin-nx : 12.3.3
  @nrwl/express : Not Found
  @nrwl/jest : 12.3.3
  @nrwl/linter : 12.3.3
  @nrwl/nest : 12.3.3
  @nrwl/next : Not Found
  @nrwl/node : 12.3.3
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.3.3
  @nrwl/web : Not Found
  @nrwl/workspace : 12.3.3
  @nrwl/storybook : Not Found
  @nrwl/gatsby : Not Found
  typescript : 4.2.4
```

**Comment by yharaskrik** [comment]
Going to do some more testing but I think this has been fixed in either 12.3.4 or 12.3.5

**Comment by AgentEnder** [comment]
Confirmed fixed, tested in 12.4
